### PR TITLE
Scheduled backup

### LINKED ISSUE #361: Feature request: scheduled export
I'd like to create an automated export schedule - ideally being able to set it to occur every N days at X time.

### PR DESCRIPTION
**Pull request by Merkost**
Fix #361 
Please, also check this pull request https://github.com/SimpleMobileTools/Simple-Commons/pull/1739

### CODE REVIEW

**Comment by Merkost** [comment]
Feature request from me: Add ability to choose the period of backup

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt [hunk 1]
```diff
@@ -12,4 +13,24 @@ class Config(context: Context) : BaseConfig(context) {
     var showTabs: Int
         get() = prefs.getInt(SHOW_TABS, ALL_TABS_MASK)
         set(showTabs) = prefs.edit().putInt(SHOW_TABS, showTabs).apply()
+
+    var autoBackup: Boolean
+        get() = prefs.getBoolean(AUTO_BACKUP, false)
+        set(enableAutomaticBackups) = prefs.edit().putBoolean(AUTO_BACKUP, enableAutomaticBackups).apply()
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt:19]: make variable names consistent, either enableAutomaticBackups or autoBackup

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt [hunk 2]
```diff
@@ -12,4 +13,24 @@ class Config(context: Context) : BaseConfig(context) {
...
+
+    var autoBackupFolder: String
+        get() = prefs.getString(AUTO_BACKUP_FOLDER, Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath)!!
+        set(autoBackupPath) = prefs.edit().putString(AUTO_BACKUP_FOLDER, autoBackupPath).apply()
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt:23]: here too, do not mix autoBackupPath and autoBackupFolder. In other places too.

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt (hunk 1 above)
Comment (Merkost) [reply app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt:19]: I've copied it from calendar, then we need to change it there too, or just extract it to commons

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/contacts/pro/helpers/Config.kt]: right, it has to be consistent with other apps too, update the calendar please

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/dialogs/ManageAutoBackupsDialog.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                activity.setupDialogStuff(view, this, R.string.manage_automatic_backups) { dialog ->
+                    dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+                        if (view.backup_contact_sources_list.adapter == null) {
+                            return@setOnClickListener
+                        }
+                        val filename = view.backup_contacts_filename.value
+                        when {
+                            filename.isEmpty() -> activity.toast(R.string.empty_name)
+                            filename.isAValidFilename() -> {
+                                val file = File(backupFolder, "$filename.ics")
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/dialogs/ManageAutoBackupsDialog.kt:79]: contacts use .vcf, not .ics

File: app/src/main/res/layout/dialog_manage_automatic_backups.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        </com.simplemobiletools.commons.views.MyTextInputLayout>
+
+        <com.simplemobiletools.commons.views.MyTextInputLayout
+            android:id="@+id/backup_contacts_filename_hint"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/activity_margin"
+            android:layout_marginEnd="@dimen/activity_margin"
+            android:hint="@string/filename_without_json"
```
Comment (tibbi) [inline app/src/main/res/layout/dialog_manage_automatic_backups.xml:38]: replace json with vcf

**Comment by tibbi** [comment]
lets enable all contact sources by default

**Comment by tibbi** [comment]
should be good, thanks

**Pull request merged**